nilearn/input_data/nifti_masker.py

```diff
@@ -200,9 +201,62 @@ def __init__(self, mask_img=None, sessions=None, smoothing_fwhm=None,
         self.memory = memory
         self.memory_level = memory_level
         self.verbose = verbose
+        self.reports = reports
+        self._report_description = ('This report shows the input Nifti '
+                                    'image overlaid with the outlines of the '
+                                    'mask. We recommend to inspect the report '
+                                    'for the overlap between the mask and its '
+                                    'input image. ')
+        self._overlay_text = ('\n To see the input Nifti image before '
+                              'resampling, hover over the displayed image.')
 
         self._shelving = False
 
+    def _reporting(self):
+        """
+        Returns
+        -------
+        displays : list
+            A list of all displays to be rendered.
+        """
+        from .. import plotting
+
+        img, mask = self.input_
+        try:
+            dim = image.load_img(img).shape
+            if len(dim) == 4:
+                # compute middle image from 4D series for plotting
+                img = image.index_img(img, dim[-1] // 2)
```

Comment:
I was thinking to take the middle slice since it should give a reasonable approximation of a "typical" image in the time series, whereas summary stats could be influenced by e.g. high motion volumes. But happy to update if you think the mean makes more sense !

Reply:
that makes sense, let's keep the middle slice